make.paws/R/xmlutil.R

```diff
@@ -1,16 +1,150 @@
-# Unmarshal `data` provided as XML into the shape in `interface`.
+# Decode raw bytes XML into an R list object.
+decode_xml <- function(raw) {
+  obj <- xml_to_list(rawToChar(raw))
+  return(obj)
+}
+
+# Convert an XML string to an R list.
+xml_to_list <- function(value) {
+  if (is_empty(value)) return(NULL)
+  result <- xml2::as_list(xml2::read_xml(value))
+  return(result)
+}
+
+# Convert list to XML text
+xml_list_to_character <- function(value) {
```

Comment:
1. This should probably be `list_to_xml` since it's doing the reverse of `xml_to_list`.
2. It inserts newlines and an extra xml tag, which you can get rid of with `as.character(xml, options = "no_declaration")`. See [write_xml](https://www.rdocumentation.org/packages/xml2/versions/1.2.0/topics/write_xml).

Reply:
addressed in last commit